### PR TITLE
Fix `QueryEditorExtensions` not showing on initial render

### PR DESCRIPTION
**Pull request by joshuali925**
### Description

This PR 
1. adds a ready state for `QueryEditorExtensions` in `QueryEditor` to force a rerender when component refs are ready
2. ~replaces datasource check in query assist from `DEFAULT_QUERY.DATASET.DATASOURCE.TYPE` to hard coded `OpenSearch`~
   - this is needed because `DEFAULT_QUERY.DATASET.DATASOURCE.TYPE` is now `OPENSEARCH`. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1cde3b6e14f7af66e7a51949e35387ada39b8ba4/src/plugins/data/common/constants.ts#L52-L53 it no longer matches [`DataSource.type`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1cde3b6e14f7af66e7a51949e35387ada39b8ba4/src/plugins/data/common/datasets/types.ts#L17-L18) which is actually `OpenSearch` https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1cde3b6e14f7af66e7a51949e35387ada39b8ba4/src/plugins/data_source/common/data_sources/types.ts#L58-L59

## Changelog
- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx [hunk 1]
```diff
@@ -30,8 +30,8 @@ const getAvailableLanguages$ = (
     distinctUntilChanged(),
     switchMap(async (query) => {
       // currently query assist tool relies on opensearch API to get index
-      // mappings, external data source types (e.g. s3) are not supported
-      if (query.dataset?.dataSource?.type !== DEFAULT_QUERY.DATASET.DATASOURCE.TYPE) return [];
+      // mappings, only {@link DataSourceEngineType.OpenSearch} is supported.
+      if (query.dataset?.dataSource?.type !== 'OpenSearch') return [];
```
Comment (kavilla) [inline src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx:34]: i actually just end up just modifying this: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7852

Comment (joshuali925) [reply src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx:34]: oh nice i'll revert mine

**Comment by kavilla** [comment]
heads up: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7860

**Comment by joshuali925** [comment]
closing this, not sure if this is needed but right now query assist works on main

**Pull request closed without merging**